### ISSUE TITLE
Slate transform for thematic breaks

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/markdown-transform#99]
**Describe the bug**
The Slate transform turns thematic seems to turn thematic breaks into empty paragraphs.

**To Reproduce**

Here is a simple `test.md`:

```
This is a heading
----

This is a thematic break:

----

This is more paragraph
```

The Cicero DOM looks fine:

```
bash-3.2$ markus parse --sample test.md --cicero 
10:38:13 - info:
{
  "$class": "org.accordproject.commonmark.Document",
  "xmlns": "http://commonmark.org/xml/1.0",
  "nodes": [
    {
      "$class": "org.accordproject.commonmark.Heading",
      "level": "2",
      "nodes": [
        {
          "$class": "org.accordproject.commonmark.Text",
          "text": "This is a heading"
        }
      ]
    }, 
    {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": [
        {
          "$class": "org.accordproject.commonmark.Text",
          "text": "This is a thematic break:"
        }
      ]
    }, 
    {
      "$class": "org.accordproject.commonmark.ThematicBreak"
    }, 
    {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": [
        {
          "$class": "org.accordproject.commonmark.Text",
          "text": "This is more paragraph"
        }
      ]
    }
  ]
}
```

But in the Slate DOM the thematic break has disappeared:
```
bash-3.2$ markus parse --sample test.md --slate
10:38:37 - info:
{
  "object": "value",
  "document": {
    "object": "document",
    "data": {
    },
    "nodes": [
      {
        "object": "block",
        "type": "heading_two",
        "data": {
        },
        "nodes": [
          {
            "object": "text",
            "text": "This is a heading",
            "marks": []
          }
        ]
      }, 
      {
        "object": "block",
        "type": "paragraph",
        "data": {
        },
        "nodes": [
          {
            "object": "text",
            "text": "This is a thematic break:",
            "marks": []
          }
        ]
      }, 
      {
        "object": "block",
        "type": "paragraph",
        "data": {
        },
        "nodes": []
      }, 
      {
        "object": "block",
        "type": "paragraph",
        "data": {
        },
        "nodes": [
          {
            "object": "text",
            "text": "This is more paragraph",
            "marks": []
          }
        ]
      }
    ]
  }
}
```

**Comment by jeromesimeon** [comment]
Does anyone working on the markdown editor know what the correct Slate DOM should be? (@irmerk @DianaLease @dselman )

**Comment by jeromesimeon** [comment]
There is a reverse mapping from Slate to CiceroMark which can help since it indicates the kind of Slate block is expected: https://github.com/accordproject/markdown-transform/blob/90cf0cfa38582e9a38029573ad7fddba8470f32c/packages/markdown-slate/src/slateToCiceroMarkDom.js#L70

**Comment by jeromesimeon** [comment]
The CiceroMark to Slate DOM transform wrongly creates a paragraph here:
https://github.com/accordproject/markdown-transform/blob/90cf0cfa38582e9a38029573ad7fddba8470f32c/packages/markdown-slate/src/ToSlateVisitor.js#L221

**Comment by jeromesimeon** [comment]
Fix is in, PR #102 

I am leaving this issue open until the fix can be tested inside the `markdown-editor`.

**Comment by jeromesimeon** [comment]
Confirmed working in https://github.com/accordproject/markdown-editor/pull/141
